File: MyFriends/MyFriends/ViewController.m
//
//  ViewController.m
//  MyFriends
//
//  Created by Michael McChesney on 1/22/15.
//  Copyright (c) 2015 Max McChesney. All rights reserved.
//
//Homework :
//- new app named my friends
//- uses parse (save user with username and password)
//- tableviewcontroller that shows friends (user query... this is special case look into how to query user's)
//- have a column on user class "friends" that is an array (when a friend is selected, add them to currentuser's friend array)

#import "ViewController.h"
#import <Parse/Parse.h>
#import "friendsTVC.h"
#import "CurrentFriendsTVC.h"
IB_DESIGNABLE
@interface ViewController () <UITextFieldDelegate>

@property (weak, nonatomic) IBOutlet UITextField *usernameTextField;
@property (weak, nonatomic) IBOutlet UITextField *passwordTextField;
@property (weak, nonatomic) IBOutlet UILabel *titleLabel;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    [self.titleLabel setTransform:CGAffineTransformMakeRotation(-M_PI / 2)];
    
//    NSAttributedString *usernamePlaceholder = [NSAttributedString alloc] initWithString:@"Username" attributes:<#(NSDictionary *)#>
    
//    self.usernameTextField.attributedPlaceholder = @"Username";
//    self.passwordTextField.attributedPlaceholder = @"Password";
    self.passwordTextField.delegate = self;
    self.usernameTextField.delegate = self;
}

-(BOOL)textFieldShouldReturn:(UITextField *)textField {

    [self signUpUser:self];     // THIS RIGHT?
    
    return YES;
}


- (IBAction)signUpUser:(id)sender {
    
    [PFUser currentUser].username = self.usernameTextField.text;
    [PFUser currentUser].password = self.passwordTextField.text;
//    [PFUser logInWithUsername:self.usernameTextField.text password:self.passwordTextField.text];
    [[PFUser currentUser] saveInBackground];
    
//    UITabBarController *tabBarController = [[UITabBarController alloc] init];
    UITabBarController *tabBarController = [self.storyboard instantiateViewControllerWithIdentifier:@"tabBarController"];
    
//    friendsTVC * friendsView = [[friendsTVC alloc] init];
    [self presentViewController:tabBarController animated:YES completion:nil];

    
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: Archive/Archive/ViewController.swift
//
//  ViewController.swift
//  Archive
//
//  Created by Michael McChesney on 2/26/15.
//  Copyright (c) 2015 Max McChesney. All rights reserved.
//

import UIKit

private let documentsDirectoryPath = NSSearchPathForDirectoriesInDomains(NSSearchPathDirectory.DocumentDirectory, NSSearchPathDomainMask.AllDomainsMask, true)[0] as String

class ViewController: UIViewController {
    
    let dotsData = documentsDirectoryPath + "/dots.data"

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
        println(documentsDirectoryPath)
        
        loadDots()
        
        NSNotificationCenter.defaultCenter().addObserverForName("appIsClosing", object: nil, queue: NSOperationQueue.mainQueue()) { (notification) -> Void in
            
            self.saveDots()
            
        }
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


    func saveDots() {
        
        // get dots from boardview
        // archive dots to file path
        
        let bView = self.view as BoardView
        
        NSKeyedArchiver.archiveRootObject(bView.dots, toFile: dotsData)
        
    }
    
    func loadDots() {
        
        // get file path for dots archive
        // unarchive file path to dots array
        // set boardview dots
        
        let bView = self.view as BoardView
        
        println(NSFileManager.defaultManager().fileExistsAtPath(dotsData))
        
        if let dots = NSKeyedUnarchiver.unarchiveObjectWithFile(dotsData) as? [Dot] {
            
            bView.dots = dots
            bView.setNeedsDisplay()
            
        }
        
    }
    
    
    @IBAction func clearDots(sender: AnyObject) {
        
        let bView = self.view as BoardView
        bView.dots = []
        
        NSKeyedArchiver.archiveRootObject(bView.dots, toFile: dotsData)
        
        bView.setNeedsDisplay()
    }
    
    
}

